Code/GraphMol/Atom.cpp

```diff
@@ -413,8 +413,14 @@ int calculateExplicitValence(const Atom &atom, bool strict, bool checkIt) {
       maxValence = ovalens.back();
       offset -= atom.getFormalCharge();
     }
+    // we have historically accepted two-coordinate [H-] as a valid atom. This
+    // is highly questionable, but changing it requires some thought. For now we
+    // will just keep accepting it
+    if (atom.getAtomicNum() == 1 && atom.getFormalCharge() == -1) {
+      maxValence = 2;
+    }
```

Comment:
I also stumbled on `H-` while testing my PR and was a bit surprised. I am not quite sure why we do this, but I may be missing the historical background. Are you sure you need the above change, though? I do not think there were any tests failing before your proposed change.

Reply:
> I also stumbled on `H-` while testing my PR and was a bit surprised. I am not quite sure why we do this, but I may be missing the historical background. Are you sure you need the above change, though? I do not think there were any tests failing before your proposed change.

A bunch of tests (I think 6 or 7) were failing if I didn't add this. I started fixing them instead of adding this workaround, but then realized that it was maybe a step too far, so I added the "let's keep accepting `C[H-]C` for now" workaround.

I think it's worth coming back to this at some point since it doesn't fit the way we normally do things, but that would be for the 2025.03.1 release.

Reply:
I'm sure some solar chemist will want their hydride, but I, for one, can do without.  :)

Reply:
> I'm sure some solar chemist will want their hydride, but I, for one, can do without. :)

It's not just a hydride, it's a hypervalent hydride!

Reply:
> A bunch of tests (I think 6 or 7) were failing if I didn't add this. I started fixing them instead of adding this workaround, but then realized that it was maybe a step too far, so I added the "let's keep accepting `C[H-]C` for now" workaround.

I see; I also started doing exactly the same with the tests and then thought you would have hated that (you are always in my thoughts), so I also put in some countermeasure :)